content/posts/2024-11-05-flang-new.md

```diff
@@ -0,0 +1,444 @@
+
+IBM has an excellent [article](https://www.ibm.com/history/fortran) on the history
+of Fortran and I encourage you to look at the
+["Programmer's Primer for Fortran"](https://archive.computerhistory.org/resources/text/Fortran/102653987.05.01.acc.pdf)
+if you want to know more.
+
+The name Fortran came from "Formula Translation" and reading the Programmer's
+Primer you will see why. A focus on the mathematical use case, freeing programmers
+from writing machine specific assembly code.
+
+Instead they could write code that looked like their formulas. Something we expect
+today but at the time was a revolution. This focus led to Fortran being used
+heavily in scientific computing. Weather modelling, fluid dynamics and
+computational chemistry, just to name a few.
+
+A 2023 [study](https://arxiv.org/pdf/2308.13274) put it this way:
+> Whilst many alternative programming languages have come
+> and gone, it [Fortran] has regained its popularity for writing high
+> performance codes. Indeed, over 80% of the applications
+> running on ARCHER2, a 750,000 core Cray EX which is
+> the UK national supercomputer, are written in Fortran.
+
+Fortran has had a [resurgency](https://ondrejcertik.com/blog/2021/03/resurrecting-fortran/)
+in recent years. Gaining a [package manager](https://fpm.fortran-lang.org/), a de facto
+[standard library](https://github.com/fortran-lang/stdlib) and [LFortran](https://lfortran.org/),
+a compiler that enables interactive Fortran programming (LFortran also uses LLVM).
+
+If you are interested in learning more about Fortran, [fortran-lang.org](https://fortran-lang.org/)
+is a great place to start.
+
+# Did We Need Another Fortran Compiler?
+
+There are many Fortran compilers. Some are vendor specific such as the
+[Intel Fortran Compiler](https://www.intel.com/content/www/us/en/developer/tools/oneapi/fortran-compiler.html)
+or NVIDIA's [HPC compilers](https://developer.nvidia.com/hpc-compilers) and
+there are open source options like [gfortran](https://gcc.gnu.org/fortran/).
+
+So why build another one? In 2020 Gary Klimowicz (NVIDIA)
+[laid out](https://www.youtube.com/watch?v=Fy68k5hHgLk) the goals for Flang,
+paraphrased here:
+
+* Develop Flang under a permissive license like LLVM's, which is more palatable
+  to commercial users and contributors.
+* Develop an active community of Fortran compiler developers that includes
+  companies and institutions.
+* Support Fortran tool development by being based on existing LLVM frameworks.
+* Support Fortran language experimentation for future language standards proposals.
+
+The last one can be explained by comparison to Clang. For C++ feature, a common
+path is to implement them on top of Clang, GCC, or both, to prove the feature can work.
+Implementing the feature multiple times in different compilers uncovers any
+assumptions in the design that may be a problem for one compiler or another.
+
+In the same way, Flang and gfortran can provide that diversity.
```

Comment:
```suggestion
In the same way, Flang and GFortran can provide that diversity.
```

Reply:
It would help if GNU would stick to their own branding (https://gcc.gnu.org/wiki/GFortran) but yes, I will fix all the instances of this.

Reply:
Fixed.